### PR TITLE
Allow quick new card with minimal options

### PR DESCRIPTION
**Pull request by gizzmojr**
Populate modal with default op_sys, platform, priority, severity
If op_sys and/or platform null, set to ALL (BZ cgi auto determines through browser)

Check for disabled components and versions, so don't list them as options.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -199,6 +199,82 @@
...
+                background-color: #fefefe;
+                margin: auto;
+                flex-direction: column;
+                align-items: flex-start;
+                width: 30%;
+                box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2),0 6px 20px 0 rgba(0,0,0,0.19);
+                -webkit-animation-name: animatetop;
+                -webkit-animation-duration: 0.4s;
+                animation-name: animatetop;
+                animation-duration: 0.4s
```
Comment (leif81) [inline index.html:224]: Probably safe to remove the `-webkit-` prefix ones.

File: index.html [hunk 2]
```diff
@@ -199,6 +199,82 @@
...
+            }
+            @-webkit-keyframes animatetop {
+                from {top:-300px; opacity:0}
+                to {top:0; opacity:1}
+            }
```
Comment (leif81) [inline index.html]: Here too. Could be removed safely I think.

File: index.html [hunk 3]
```diff
@@ -241,6 +317,32 @@
...
         </div>
+        <div id="modalAddBug" class="modal">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <span class="modalClose">x</span>
+                    <h2>Add New Bug</h2>
+                </div>
+                <div class="modal-body">
+                    <label>Summary of the issue</label>
+                    <input type="text" required id="textAddBugSummary" name="summary" value=""</input>
```
Comment (leif81) [inline index.html]: `label`'s usually wrap an `input` element.

File: index.html [hunk 4]
```diff
@@ -241,6 +317,32 @@
...
+                    <label>Pick a component</label>
+                    <select id="textComponent" name="component" placeholder="Component" value=""></select>
+                    <label>Pick a version</label>
+                    <select id="textVersion" name="version" placeholder="Version"></select>
+                    <label>Using milestone</label>
```
Comment (leif81) [inline index.html]: Can we put the milestone value in the modal title instead? e.g. "Add new bug to milestone XYZ"

File: index.html [hunk 5]
```diff
@@ -561,6 +665,38 @@
...
+                    versions.forEach(function(version) {
+                        if (!(version.is_active)) { return }
+                        var opt = document.createElement('option');
+                        opt.innerText = version.name;
+                        opt.value = version.name;
+                        document.getElementById("textVersion").appendChild(opt);
+                    });
+                });
+            }
+
```
Comment (leif81) [inline index.html]: Not sure how to reproduce it, but I once saw the same component names and versions listed twice in the select boxes in the modal. Just FYI.

File: index.html [hunk 6]
```diff
@@ -241,6 +317,32 @@
...
+        <div id="modalAddBug" class="modal">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <span class="modalClose">x</span>
+                    <h2>Add New Bug</h2>
+                </div>
+                <div class="modal-body">
+                    <label>Summary of the issue</label>
+                    <input type="text" required id="textAddBugSummary" name="summary" value=""</input>
+                    <label>Pick a component</label>
```
Comment (leif81) [inline index.html]: Can we remove the "Pick a" part to be shorter.

File: index.html [hunk 7]
```diff
@@ -241,6 +317,32 @@
...
+                    <select id="textVersion" name="version" placeholder="Version"></select>
+                    <label>Using milestone</label>
+                    <input type="text" id="textModalMilestone" name="milestone" value="" readonly></input>
+                    <label>Using Bugzilla defaults</label>
+                    <ul style="list-style-type:circle">
+                        <li>op_sys = <label id="labelOpSys"></label></li>
+                        <li>platform = <label id="labelPlatform"></label></li>
+                        <li>priority = <label id="labelPriority"></label></li>
+                        <li>severity = <label id="labelSeverity"></label></li>
+                    </ul>
```
Comment (leif81) [inline index.html:337]: I'd be OK even hiding this to keep the dialog really low on words.

**Review by leif81 — COMMENTED** [review]
Really nice.

Most of my comments were about slimming down the content of the modal to keep it fast and light feeling instead of the traditional dialog that gets so wordy.

Looking forward to re-using the modal logic soon for adding comments.

**Comment by leif81** [comment]
Also, not sure how to reproduce it, but I once saw the same component names and versions listed twice in the select boxes in the modal. Just FYI.

File: index.html (hunk 3 above)
Comment (leif81) [reply index.html]: Also there's a missing `>` there on the `input`.

**Comment by leif81** [comment]
I think we should also show the description field in the modal. If we limit the modal to just the mandatory fields (summary, description, component, version) we'll be still well ahead of the traditional UI in terms of lighterweight.

**Comment by leif81** [comment]
I also have some really minor styling tweaks I'd like to try on the modal, but those can wait until after this is merged.

File: index.html (hunk 2 above)
Comment (gizzmojr) [reply index.html]: It was some example code I found. I should of also sighted where I got it from/author. 

That said, you're suggestion is correct. I'll clean this up.

File: index.html (hunk 6 above)
Comment (gizzmojr) [reply index.html]: Sure

File: index.html (hunk 7 above)
Comment (gizzmojr) [reply index.html:337]: Sure, so if someone wanted them back it be easy to show them

**Comment by gizzmojr** [comment]
Ok all wrapped up. Great suggestions.

**Comment by leif81** [comment]
Very nice :+1: 

:ship: :dancing_women:

**Pull request merged**